modules/docker/Makefile.build

```diff
@@ -0,0 +1,42 @@
+## Use DOCKER_IMAGE_NAME envvar to specify docker image with tags
+## User ARGS to pass arguments
+DOCKER_BUILD_PATH ?= .
+DOCKER_BUILD_FLAGS ?= --pull --cache-from $(DOCKER_IMAGE_NAME)
```

Comment:
`--cache-from` is for external caches. As far as I can tell it doesn't work without the repository name, the image name, and specific tag name (e.g. `gcr.io/onec-co/built-object-service:master`).

Otherwise, I see an error like this:
```
 => ERROR importing cache manifest from built-object-service                                                                                                                                                      
```

The above comment (`# Use DOCKER_IMAGE_NAME envvar to specify docker image with tags`) seems contradictory to how the `DOCKER_IMAGE_NAME` env var is used here:
(does not appear to be intended to include the repo name and a tag)
https://github.com/oneconcern/build-harness/blob/5392d1b170c2ecfd8288bd5d0bc28bc5cc521802/modules/docker/Makefile.build#L25
https://github.com/oneconcern/build-harness/blob/5392d1b170c2ecfd8288bd5d0bc28bc5cc521802/modules/docker/Makefile.build#L33
https://github.com/oneconcern/build-harness/blob/5392d1b170c2ecfd8288bd5d0bc28bc5cc521802/modules/docker/Makefile.build#L42

I think it might make sense to have a separate optional Makefile variable (e.g. `DOCKER_CACHE_IMAGE`) that could be set externally (outside of the build-harness) to optionally set an image/tag to use for the `--cache-from` option.

Reply:
Actually, you might want to do this (if you want it to be optional):
```
ifdef DOCKER_CACHE_IMAGE
    DOCKER_BUILD_FLAGS ?= --pull --cache-from $(DOCKER_CACHE_IMAGE)
else
    DOCKER_BUILD_FLAGS ?= --pull
endif
```

Reply:
Updated to make `DOCKER_CACHE_IMAGE` optional.